syntax = "proto3";

package receiver;

// Define the request and response messages
message MyMethodRequest {
    string metadata = 1;
    string text = 2;
}

message MyMethodResponse {
    string message = 1;
}

message JsonMethodRequest {
    string json_data = 1;
}

message JsonMethodResponse {
    string received = 1;
    string message = 2;
}

// Define the service
service ReceiverService {
    rpc MyMethod(MyMethodRequest) returns (MyMethodResponse);
    rpc JsonMethod(JsonMethodRequest) returns (JsonMethodResponse);
}